ydb/core/tx/schemeshard/schemeshard__operation_alter_login.cpp

```diff
@@ -51,12 +75,47 @@ class TAlterLogin: public TSubOperationBase {
                 }
                 case NKikimrSchemeOp::TAlterLogin::kModifyUser: {
                     const auto& modifyUser = alterLogin.GetModifyUser();
-                    auto response = context.SS->LoginProvider.ModifyUser({.User = modifyUser.GetUser(), .Password = modifyUser.GetPassword()});
+
+                    NLogin::TLoginProvider::TModifyUserRequest request;
+
+                    request.User = modifyUser.GetUser();
+
+                    switch (modifyUser.GetCanLogin()) {
+                        case NKikimrSchemeOp::ETypeOfLogin::Login:
+                        {
+                            request.CanLogin = NLogin::TLoginProvider::ETypeOfLogin::Login;
+                            break;
+                        }
+                        case NKikimrSchemeOp::ETypeOfLogin::NoLogin:
+                        {
+                            request.CanLogin = NLogin::TLoginProvider::ETypeOfLogin::NoLogin;
+                            break;
+                        }
+                        case NKikimrSchemeOp::ETypeOfLogin::Undefined:
+                        {
+                            request.CanLogin = NLogin::TLoginProvider::ETypeOfLogin::Undefined;
+                            break;
+                        }
+                    }
+
+                    if (modifyUser.HasPassword()) {
+                        request.Password = modifyUser.GetPassword();
+                    }
+
+                    auto response = context.SS->LoginProvider.ModifyUser(request);
                     if (response.Error) {
                         result->SetStatus(NKikimrScheme::StatusPreconditionFailed, response.Error);
                     } else {
                         auto& sid = context.SS->LoginProvider.Sids[modifyUser.GetUser()];
-                        db.Table<Schema::LoginSids>().Key(sid.Name).Update<Schema::LoginSids::SidType, Schema::LoginSids::SidHash>(sid.Type, sid.Hash);
+
+                        if (modifyUser.HasPassword()) {
+                            db.Table<Schema::LoginSids>().Key(sid.Name).Update<Schema::LoginSids::SidType, Schema::LoginSids::SidHash>(sid.Type, sid.Hash);
+                        }
+
+                        if (request.CanLogin != NLogin::TLoginProvider::ETypeOfLogin::Undefined) {
+                            db.Table<Schema::LoginSids>().Key(sid.Name).Update<Schema::LoginSids::SidType, Schema::LoginSids::IsEnabled>(sid.Type, sid.IsEnabled);
+                        }
```

Comment:
May be we do not need to check `modifyUser.HasPassword()` and `request.CanLogin != NLogin::TLoginProvider::ETypeOfLogin::Undefined` and just write all parameters from sid to local db?

```suggestion
db.Table<Schema::LoginSids>().Key(sid.Name).Update<Schema::LoginSids::SidType, Schema::LoginSids::SidHash, Schema::LoginSids::IsEnabled>(sid.Type, sid.Hash, sid.IsEnabled);
```